### PR TITLE
Fix remaining issues in 0.42

### PR DESCRIPTION
**Pull request by iverc**
- Fix remaining warnings
- Fix unit tests & related issues
- Pull in docker fixes (integration tests fixees)

### CODE REVIEW

**Review by intl-man — APPROVED** [review]
LGTM.

File: x/dispensation/utils/parser_test.go [hunk 1]
```diff
@@ -77,3 +77,43 @@ func TestParseOutput(t *testing.T) {
 	assert.NoError(t, err)
 	assert.Equal(t, len(outputs), count)
 }
+
+func TestSplitBetweenReciepients(t *testing.T) {
```
Comment (Jedi2002) [inline x/dispensation/utils/parser_test.go:81]: We can remove this , its not relevent any more

**Review by Jedi2002 — COMMENTED** [review]
Looks great ,  minor comment which should be addressed before merging

File: x/oracle/genesis.go [hunk 2]
```diff
@@ -11,20 +11,21 @@ import (
...
-
-	var wl []sdk.ValAddress
-	for _, addr := range data.AddressWhitelist {
-		if len(strings.TrimSpace(addr)) == 0 {
-			continue
+	if data.AddressWhitelist != nil {
+		wl := make([]sdk.ValAddress, len(data.AddressWhitelist))
+		for i, entry := range data.AddressWhitelist {
+			if len(strings.TrimSpace(entry)) == 0 {
+				continue
```
Comment (timlind) [inline x/oracle/genesis.go:18]: This introduces a bug where wl has an zero value entry created by make() but not assigned here, which is why the code was using an unallocated slice.

**Review by timlind — CHANGES_REQUESTED** [review]

**Review by timlind — APPROVED** [review]

**Pull request merged**